flora/flora.ino

```diff
@@ -100,123 +189,190 @@ bool getSensorData(BLEAddress pAddress, bool getBattery) {
   }
   Serial.println(" ");
 
-  float temp = (val[0] + val[1] * 256) / ((float)10.0);
+  float temperature = (val[0] + val[1] * 256) / ((float)10.0);
+  Serial.print("-- Temperature: ");
+  Serial.println(temperature);
+
   int moisture = val[7];
+  Serial.print("-- Moisture: ");
+  Serial.println(moisture);
+
   int light = val[3] + val[4] * 256;
+  Serial.print("-- Light: ");
+  Serial.println(light);
+ 
   int conductivity = val[8] + val[9] * 256;
-  
-  char buffer[64];
-  
-  Serial.print("Temperature: ");
-  Serial.println(temp);
-  snprintf(buffer, 64, "%f", temp);
-  client.publish(MQTT_TEMPERATURE, buffer);
-  
+  Serial.print("-- Conductivity: ");
+  Serial.println(conductivity);
 
-  Serial.print("Moisture: ");
-  Serial.println(moisture);
-  snprintf(buffer, 64, "%d", moisture);
-  client.publish(MQTT_MOISTURE, buffer);
+  if (temperature > 200) {
+    Serial.println("-- Unreasonable values received, skip publish");
+    return false;
+  }
 
-  Serial.print("Light: ");
-  Serial.println(light);
-  snprintf(buffer, 64, "%d", light);
-  client.publish(MQTT_LIGHT, buffer);
+  char buffer[64];
 
-  Serial.print("Conductivity: ");
-  Serial.println(conductivity);
+  snprintf(buffer, 64, "%f", temperature);
+  client.publish((baseTopic + "temperature").c_str(), buffer); 
```

Comment:
we could carve out these formatting & publishing instructions into a separate function... it looks too repetitive :-) Anyway, there's no real need to do this in this PR.

Reply:
:+1: